ballista/executor/src/executor_server.rs

```diff
@@ -372,18 +379,78 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> ExecutorServer<T,
         )?)
     }
 
-    async fn run_task(
+    async fn decode_task(
```

Comment:
These are the main functional changes (everything else is jsut cleaning up logging). There are basically 4 places we can fail when running a task (outside of the task execution itself failing which is already handled). 
1. Creating the configs from the serialized key/value properties
2. Decoding the execution plan
3. Registering the circuit breaker
4. De-registering the circuit breaker.

This should ensure if any one of those 4 fail then we still send a task status back to the scheduler.